optimum/fx/quantization/quantizer.py

```diff
@@ -0,0 +1,103 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional, Union
+
+import torch
+from torch.ao.quantization.quantize_fx import convert_fx
+from transformers.optimization import AdamW
+
+from ...utils.runs import QuantizationApproach
+from .functions import calibrate, calibrate_qat, prepare_fx, prepare_qat_fx
+
+
+if TYPE_CHECKING:
+    from datasets import Dataset
+    from torch.ao.quantization.quantize_fx import ObservedGraphModule
+    from transformers import PreTrainedModel
+
+
+def quantize(
+    model: "PreTrainedModel",
+    approach: Union[str, QuantizationApproach],
+    qconfig_dict: Dict[str, Any],
+    preprocess_func: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+    calibration_dataset: Optional["Dataset"] = None,
+    num_calibration_samples: int = -1,
+    input_names: Optional[List[str]] = None,
+) -> torch.nn.Module:
+    """
+    Handles the quantization of the model end-to-end, by taking care of all the intermediate steps.
+
+    Args:
+        model (`transformers.PreTrainedModel`):
+            The model to quantize.
+        approach (`str` or [`~optimum.utils.runs.QuantizationApproach`]):
+            The quantization approach.
+        qconfig_dict (`Dict[str, Any]`):
+            The dictionary specifying how each part of the model should be quantized.
+            Please refer to [the PyTorch documentation](https://pytorch.org/docs/stable/generated/torch.quantization.quantize_fx.prepare_fx.html#torch.quantization.quantize_fx.prepare_fx)
+            for more details.
+        preprocess_func (`Callable[[Dict[str, Any]], [Dict[str, Any]], *optional*):
+            The preprocessing function to apply to the calibration dataset to make the data ready to be fed to the
```

Comment:
We should document / explicit how the `preprocess_func` is called, and have examples in the doc.